cli/src/main/java/org/jboss/galleon/cli/HelpSupport.java

```diff
@@ -168,6 +89,163 @@ private static String pad(int max, String k) {
         return lst;
     }
 
+    public static String buildHelp(CommandRegistry<? extends Command, ? extends CommandInvocation> registry,
+                                    Set<String> commands) throws CommandNotFoundException {
+        return buildHelp(registry, commands, true);
+    }
+    private static String buildHelp(CommandRegistry<? extends Command, ? extends CommandInvocation> registry,
+            Set<String> commands, boolean footer) throws CommandNotFoundException {
+        TreeMap<CommandDomain, Set<String>> groupedCommands = new TreeMap<>();
+        for (String command : commands) {
+            CommandDomain group = CommandDomain.getDomain(registry.getCommand(command, null).getParser().getCommand());
+            String commandTree = getCommandTree(registry, command);
+
+            if (group == null) {
+                group = CommandDomain.OTHERS;
+            }
+
+            Set<String> currentDescriptions = groupedCommands.get(group);
+
+            if (currentDescriptions == null) {
+                currentDescriptions = new TreeSet<>();
+                groupedCommands.put(group, currentDescriptions);
+            }
+
+            currentDescriptions.add(commandTree);
+        }
+
+        StringBuilder sb = new StringBuilder();
+        for (CommandDomain group : groupedCommands.keySet()) {
+            sb.append("== ").append(group.getDescription()).append(" ==");
```

Comment:
Looking at the output, a blank line before domains would help readability.

Reply:
Agreed